pkg/api/server.go

```diff
@@ -1175,6 +1186,7 @@ func filterObjectsByFields(object runtime.Object, selector fields.Selector) runt
 	case *corev1.EventList:
 		filtered := &corev1.EventList{}
 		for _, item := range o.Items {
+			item := item
```

Comment:
Is it necessary since there is no goroutine inside the loop?

Reply:
> Is it necessary since there is no goroutine inside the loop?

With this specific code, its not necessary. Just being defensive in case `eventToSelectableFields(&item)` function were to try to store the `&item` variable.

Also, the for-loop variable scope behaviour is not limited to goroutines. It occurs whenever a loop variable pointer gets referenced outside or the for loop scope

```go
package main

func main() {
	s := []string{"a", "b", "c"}
	r := []*string{}

	for _, v := range s {
		// v := v for-loop var scope workaround
		r = append(r, &v)
	}

	for _, v := range r {
		println(*v)
	}
}
```

Thankfully, we won't need this workaround in [go 1.22 onwards](https://github.com/golang/go/wiki/LoopvarExperiment). Its experimental at the moment in go 1.21 (`GOEXPERIMENT=loopvar go run main.go`)